src/routes/tags/create-tag.e2e.test.js

```diff
@@ -0,0 +1,159 @@
+const { expect } = require('chai')
+const { faker } = require('@faker-js/faker')
+const { fixtures } = require('../../../test/utils')
+const createTag = require("../../services/tags/create-tag")
+
+const endpoint = "/api/v1/tags"
+
+describe('POST /tags', () => {
+
+	let user
+	let accessToken
+
+	before(async ()=> {
+
+		user = await fixtures.insertUser() 
+		const body = { id: user.id, email: user.email }
+		accessToken = fixtures.generateAccessToken(
+			{user : body}
+		)
+
+	})
+
+	describe('Failure', () => {
+
+		let tag
+		before(async ()=>{
+			tag = {
+				title : faker.random.word(5),
+				content : faker.random.words()
+			}
+		})
+	
+		it('should return 401 if request is not authenticated', async () => 
+			fixtures.api()
+				.post(endpoint)
+				.expect(401)
+		)
+        
+		it('should return 400 if title is not a string', async () => {
+			const expectedError = {
+				"error": {
+					"message": "title must be a string"
+				},
+				"status": "failed"
+			}
+			return fixtures.api()
+				.post(endpoint)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({ 
+					title : 123,
+					content : tag.content
+				})
+				.expect(400, expectedError)
+		})
+
+		it('should return 400 if content is not a string', async () => {
+			const expectedError = {
+				"error": {
+					"message": "content must be a string"
+				},
+				"status": "failed"
+			}
+			return fixtures.api()
+				.post(endpoint)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({ 
+					title : tag.title,
+					content : 123
+				})
+				.expect(400, expectedError)
+		})
+		
+		it('should return 400 if title field is empty', async () => {
+
+			const expectedError = {
+				"error": {
+					"message": "title is not allowed to be empty"
+				},
+				"status": "failed"
+			}
+			
+			fixtures.api()
+				.post(endpoint)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({
+					title : "",
+					content : tag.content
+					 })
+				.expect(400, expectedError)
+		})
+
+		it('should return 400 if tag already exists', async () => {
+			await createTag(tag)
```

Comment:
kindly use a utils function for tasks like these

by testing the endpoint, we are implicitly testing the `createTag` service function itself since that's what endpoint itself is using in the long run. 

That's the reason why we try not to call the same service in the endpoint test

Reply:
yes please